bcipy/language/model/gpt2.py

```diff
@@ -81,84 +182,122 @@ def __get_char_predictions(self, word_prefix: str) -> List[tuple]:
 
         return char_prob_tuples
 
-    def __build_vocab(self) -> Dict[int, str]:
+    def __interpolate_language_models(self, lm1: Dict[str, float], lm2: Dict[str, float], coeff: float) -> List[Tuple]:
         """
-        Build a vocabulary table mapping token index to word strings
+        interpolate two language models
+        Args:
+            lm1 - the first language model (a dict with char as keys and prob as values)
+            lm2 - the second language model (same type as lm1)
+            coeff - rescale coefficient, lm1 will be scaled by coeff and lm2 will be
+            scaled by (1-coeff)
         Response:
-            a dict that maps token index to word strings
+            a list of (char, prob) tuples representing an interpolated language model
         """
+        combined_lm = Counter()
 
-        idx_to_word = {}
-        for idx in range(self.vocab_size):
-            word = self.tokenizer.decode([idx]).strip()
-            idx_to_word[idx] = word
+        for char in lm1:
+            combined_lm[char] = lm1[char] * coeff + lm2[char] * (1 - coeff)
 
-        return idx_to_word
+        for char in lm2:
+            if char not in lm1:
+                combined_lm[char] = lm2[char] * (1 - coeff)
+
+        return list(sorted(combined_lm.items(), key=lambda item: item[1], reverse=True))
 
-    def __model_infer(self, text: str) -> List[float]:
+    def __rescale(self, lm: Dict[str, float], coeff: float):
```

Comment:
Could this be a utility function in this module? It seems like it might be more generally useful.

Reply:
I agree. Changed it to a static method.